spec/react.spec.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const assert = require('assert');
+const sinon = require('sinon');
+const React = require('react');
+const ReactTestRenderer = require('react-test-renderer');
+
+const equal = require('../index');
+const tests = require('./tests');
+
+class ChildWithShouldComponentUpdate extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    // this.props.children is a div with a circular reference to its owner, Container
+    return !equal(this.props, nextProps);
+  }
+  render() {
+    return null;
+  }
+}
+
+class Container extends React.Component {
+  render() {
+    return React.createElement(ChildWithShouldComponentUpdate, {
+      children: React.createElement('div')
+    });
+  }
+}
+
+describe('advanced', () => {
+  let warnStub = sinon.stub(console, 'warn');
+  afterEach(() => {
+    warnStub.resetHistory();
+  });
+  after(() => {
+    warnStub.restore();
+  });
+
+  describe('React', () => {
+    describe('render element (with circular references)', () => {
+      it('can be compared without warning or errors', () => {
+        const testRenderer = ReactTestRenderer.create(React.createElement(Container));
+        testRenderer.update(React.createElement(Container));
+        assert.strictEqual(warnStub.notCalled, true);
+      });
+    });
+  });
+
+  describe('warnings', () => {
+    describe('generic circular references', () => {
+      it('generate a warning but do not throw an error', () => {
```

Comment:
NIT: `generates`

Reply:
_references generate_, right?
I can switch it to _ reference generates_? is singular preferred?

Reply:
I'm cool with whatever you want.

I try to have things read like sentences starting with `it` (being the test), so maybe `it warns on circular refs, but does not throw` -> `it('warns on circular refs, but does not throw', () => {`

But, totally up to you. 😄